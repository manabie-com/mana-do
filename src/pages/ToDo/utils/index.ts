import { Todo, TodoStatus } from '../../../models/todo';
const created_date = new Date().toISOString();
export const defaultToDos = {
  content: 'Content',
  created_date,
  status: TodoStatus.ACTIVE,
  id: 'testId',
  user_id: 'firstUser',
} as Todo;

export function spawnToDos(count: number): Todo[] {
  return new Array(count).fill(defaultToDos).map(todo => {
    todo.id = `testId-${Math.random()}`;
    return todo;
  });
}
