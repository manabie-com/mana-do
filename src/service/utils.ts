import {Todo, TodoStatus} from "../models/todo";

export interface TResponse {
    status: Number,
    code: string,
    data?: any
}

export const updateTodoService = async (todoId:string, content: string): Promise<TResponse> => {

    const dataListTodo:any = await localStorage.getItem('todolist') || '[]';

    const dataToAPI = await JSON.parse(dataListTodo) as Todo[]

    const index = dataToAPI.findIndex(todo => todo.id === todoId)

    dataToAPI[index].content = content

    await localStorage.setItem('todolist', JSON.stringify(dataToAPI))

    return {status: 200, code: "success"}

}

export const removeTodoService = async (todoId: string): Promise<TResponse> => {
    const dataListTodo:any = await localStorage.getItem('todolist') || '[]';

    const dataToAPI = await JSON.parse(dataListTodo) as Todo[]

    const index = dataToAPI.findIndex(todo => todo.id === todoId)

    dataToAPI.splice(index, 1)

    await localStorage.setItem('todolist', JSON.stringify(dataToAPI))

    return {status: 200, code: "success"}

}

export const updateStatusTodoService = async (todoId:string, status: TodoStatus): Promise<TResponse> => {

    const dataListTodo:any = await localStorage.getItem('todolist') || '[]';

    const dataToAPI = await JSON.parse(dataListTodo) as Todo[]

    const index = dataToAPI.findIndex(todo => todo.id === todoId)

    dataToAPI[index].status = status

    await localStorage.setItem('todolist', JSON.stringify(dataToAPI))

    return {status: 200, code: "success"}

}