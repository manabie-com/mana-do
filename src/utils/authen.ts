import {getTokenStorage} from './storeageUtils';

export const isAuthenticated = () => Boolean(getTokenStorage());
