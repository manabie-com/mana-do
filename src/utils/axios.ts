import axios, { AxiosRequestConfig, AxiosResponse } from 'axios';
const axiosClient = axios.create({
  baseURL: 'https://61615cd137492500176312cc.mockapi.io/',
  headers: {
    'Content-Type': 'application/json',
  },
});
// Add a request interceptor
axiosClient.interceptors.request.use(
  function (config: AxiosRequestConfig) {
    // Do something before request is sent
    return config;
  },
  function (error) {
    // Do something with request error
    return Promise.reject(error);
  }
);

// Add a response interceptor
axiosClient.interceptors.response.use(
  function (response: AxiosResponse) {
    // Any status code that lie within the range of 2xx cause this function to trigger
    // Do something with response data
    return response;
  },
  function (error) {
    // Any status codes that falls outside the range of 2xx cause this function to trigger
    // Do something with response error
    return Promise.reject(error);
  }
);
export default axiosClient;