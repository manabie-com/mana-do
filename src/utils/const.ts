export const TO_DO_LIST = "TO_DO_LIST";