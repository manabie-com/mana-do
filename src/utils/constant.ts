export const LOCAL_TODOS = 'LOCAL_TODOS'; 