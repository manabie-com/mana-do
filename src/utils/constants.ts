import { TodoStatus, EnhanceTodoStatus } from "../models/todo";
export interface Tab {
  text: string;
  value: EnhanceTodoStatus;
}

export const LIST_TAB: Tab[] = [
  { text: "All", value: "ALL" },
  { text: "Active", value: TodoStatus.ACTIVE },
  { text: "Completed", value: TodoStatus.COMPLETED },
];
