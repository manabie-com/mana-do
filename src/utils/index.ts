import { Todo, TodoStatus } from '../models/todo';

export function isTodoCompleted(todo: Todo): boolean {
  return todo.status === TodoStatus.COMPLETED;
}

export function isTodoActive(todo: Todo): boolean {
  return todo.status === TodoStatus.ACTIVE;
}
