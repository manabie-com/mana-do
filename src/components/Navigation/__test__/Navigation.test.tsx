import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import { Navigation } from '../index'

import { render, fireEvent, waitFor } from "@testing-library/react"
import Service from '../../../service'
import { successResponse } from '../../../utils/constant'

import "@testing-library/jest-dom";

/**
 * Test cases
 * 1. should be rendered without crashing
 * 2. should be rendered with elements
 * 3. should be logout and redirect to /login
 */

const mockHistoryPush = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}))

describe('<Navigation />', () => {

  test('should be rendered without crashing', () => {
    const wrapper = render(<Navigation />)
    expect(wrapper.container).toMatchSnapshot()
  })

  test('should be rendered with elements', () => {
    const wrapper = render(<Navigation />)
    expect(wrapper.getByText('Logout')).toBeInTheDocument()
  })

  test('should be logout and redirect to /login', async () => {
    jest.spyOn(Service, 'logout').mockResolvedValueOnce(successResponse)
    const wrapper = render((
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>
    ))
    await waitFor(() => {
      fireEvent.click(wrapper.getByText('Logout'))
    })
    expect(mockHistoryPush).toHaveBeenCalledWith('/login');
  })

})
