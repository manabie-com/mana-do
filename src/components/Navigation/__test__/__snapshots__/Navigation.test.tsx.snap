// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should be rendered without crashing 1`] = `
<div>
  <div
    class="Navigation__container"
  >
    <a
      href="/logout"
    >
      Logout
    </a>
  </div>
</div>
`;
